engine/src/main/java/org/hibernate/search/query/engine/spi/EntityInfo.java

```diff
@@ -23,9 +22,11 @@
 
 	Object[] getProjection();
 
-	List<Integer> getIndexesOfThis();
+	Iterable<Integer> getIndexesOfThis();
```

Comment:
Why the switch to `Iterable`? After all you are changing an spi and imo an `Iterable` is a bad choice of return type.